Jenkinsfile

```diff
@@ -35,6 +35,41 @@ buildCommands.each{
         }
     }
 }
+
+def windowsbuildCommands = [
+    '"Visual Studio 15 2017 Win64" -DBUILD_SHARED_LIBS=ON ../',
```

Comment:
I personally would split up the generator (e.g. `"Visual Studio 15 2017 Win64"`) and options (e.g. `BUILD_SHARED_LIBS`).

Reason: assume I would like to introduce a new generator - e.g. `"Visual Studio 15 2017 ARM"` - then I would need two lines.